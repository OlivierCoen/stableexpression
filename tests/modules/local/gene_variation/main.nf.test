nextflow_process {

    name "Test Process GENE_VARIATION"
    script "modules/local/gene_variation/main.nf"
    process "GENE_VARIATION"
    tag "genevar"
    tag "module"

    test("Method standard deviation") {

        when {

            process {
                """
                ch_counts = Channel.fromPath( '$projectDir/tests/input/gene_variation/counts*.csv', checkIfExists: true).collect()
                ch_metadata = Channel.fromPath( '$projectDir/tests/input/gene_variation/metadata*.csv', checkIfExists: true ).collect()
                ch_mapping = Channel.fromPath( '$projectDir/tests/input/gene_variation/mapping*.csv', checkIfExists: true ).collect()
                input[0] = ch_counts
                input[1] = ch_metadata
                input[2] = ch_mapping
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
