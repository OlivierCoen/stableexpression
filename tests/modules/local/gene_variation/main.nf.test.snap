{
    "Method standard deviation": {
        "content": [
            {
                "0": [
                    "gene_variations.csv:md5,75e14dfffdef0d13f04881ad802b3a29"
                ],
                "1": [
                    [
                        "GENE_VARIATION",
                        "python",
                        "3.12.8"
                    ]
                ],
                "2": [
                    [
                        "GENE_VARIATION",
                        "polars",
                        "1.17.1"
                    ]
                ],
                "csv": [
                    "gene_variations.csv:md5,75e14dfffdef0d13f04881ad802b3a29"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-03T13:56:41.268938139"
    },
    "Method coefficient of variation": {
        "content": [
            {
                "0": [
                    "gene_variations.csv:md5,c10b715a1604d489c4dfa03aa1a9b80c"
                ],
                "1": [
                    [
                        "GENE_VARIATION",
                        "python",
                        "3.12.8"
                    ]
                ],
                "2": [
                    [
                        "GENE_VARIATION",
                        "polars",
                        "1.17.1"
                    ]
                ],
                "csv": [
                    "gene_variations.csv:md5,c10b715a1604d489c4dfa03aa1a9b80c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-03T13:56:54.967522374"
    },
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "variation_coefficients.csv:md5,53c4ce914fb26c3c0c54cf9d34f99b2b"
                ],
                "1": [
                    [
                        "GENE_VARIATION",
                        "R",
                        ""
                    ]
                ],
                "csv": [
                    "variation_coefficients.csv:md5,53c4ce914fb26c3c0c54cf9d34f99b2b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-03T11:08:35.181976931"
    }
}